pkgs/development/compilers/dotnet/combine-packages.nix

```diff
@@ -3,18 +3,29 @@ packages:
 let cli = builtins.head packages;
```

Comment:
`meta.description` and the documentation should communicate the first package is used for the CLI.

I do think that's still a bit unclear, what do you think about changing it to be a bit more explicit, by making `cli` and `packages` seperate required function arguments?

Reply:
Well, again, this is to keep backwards functionality, I believe changing this would break many already existing configurations and I'd prefer to keep it as is.

Although if it's deemed worthy of a backwards incompatible change I wouldn't be against it either.

I don't really see the issue with the current implementation as it is documented that the first package is used as the one for the binaries

Reply:
Alright, fair enough. I do think a breaking change is warranted as its not clear what SDK is used when looking purely at derivation that uses this function. You'll always want an SDK anyways.

That doesn't need to happen now though, could you maybe add a todo comment?

Reply:
I see, thing is, the runtimes also do have a CLI, although if you aren't combining SDKs I don't really see much use for this other than combining different runtimes for running different programs (although for that, `buildDotnetModule` would be more indicated).

I will add the todo comment however

Reply:
Maybe instead of adding a separate comment, we should get the SDK with the largest version from the arguments? I don't see a reason for not leaving the highest SDK as the CLI, so I think this might warrant some further discussion

Reply:
Comment added @IvarWithoutBones

Reply:
> Maybe instead of adding a separate comment, we should get the SDK with the largest version from the arguments? I don't see a reason for not leaving the highest SDK as the CLI, so I think this might warrant some further discussion

You might want to build a legacy project with an older CLIs `dotnet build`, while still having a newer version available to be used internally. AFAIK backwards incompatibilities are a thing with these. I think always using the latest version is even less clear, and also allows for less flexibility.

> I see, thing is, the runtimes also do have a CLI, although if you aren't combining SDKs I don't really see much use for this other than combining different runtimes for running different programs (although for that, buildDotnetModule would be more indicated).

I do agree that something like `buildDotnetModule` would be the better option here, but in my opinion we should provide more imperative solutions for the problem with `combinePackages` as well. We might as well give the end user the option to intuatively use it.